### DISCUSSION TITLE
How to Garnet data dump?

### DISCUSSION THREAD

**Discussion by jykim-hrc** [microsoft/garnet#426] [Q&A] [answered]
Hi
I have question.
 
I hope
GarnetServer is using database cache server on windows server.
(ex)
192.168.100.1 : WebServer
192.168.100.2 : DBServer
192.168.100.3 : GarnetServer

But GargetServer stop time(Ctrl + C) stored data is clear.
How to dump stored data??

I think situation... 😁
Server boot.
> GargetServer --dumppath "D:\GargetDB\Hello.db"
set key Hello World
... More key ....
Ctrl + C stop garnetserver
Server reboot
> GargetServer --dumppath "D:\GargetDB\Hello.db"
already add key "Hello" and more

Thank 😀

**Comment by badrishc** [comment, marked as answer]
Make sure you call `SAVE` or `BGSAVE` to save the database. When restarting Garnet, make sure you start it with `--recover true`.

**Reply by jykim-hrc** [reply]
Work! 
Document in call "Checkpoint" 😆
Thank!
